src/Exercism.Analyzers.CSharp/Analyzers/WeighingMachine/WeighingMachineAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+﻿
+using Exercism.Analyzers.CSharp.Analyzers.Shared;
+
+using static Exercism.Analyzers.CSharp.Analyzers.Shared.SharedComments;
+using static Exercism.Analyzers.CSharp.Analyzers.WeighingMachine.WeighingMachineSolution;
+
+namespace Exercism.Analyzers.CSharp.Analyzers.WeighingMachine
+{
+    internal class WeighingMachineAnalyzer : SharedAnalyzer<WeighingMachineSolution>
+    {
+        protected override SolutionAnalysis DisapproveWhenInvalid(WeighingMachineSolution solution)
+        {
+            if (solution.MissingWeighingMachineClass)
+                solution.AddComment(MissingClass(WeighingMachineClassName));
+
+            foreach (var missing in solution.MissingRequiredProperties())
+            {
+                solution.AddComment(MissingProperty(missing));
+            }
```

Comment:
I'm not sure we need this check as the tests won't even compile if a property is missing, but it also doesn't hurt :)